File: repeat_a_string/repeat_a_string.js

// JavaScript 5

// Using .repeat method
function repeatStringNumTimes(str, num) {
  "use strict";
  // if num is not a positive number
  if (num < 0) {
    return "";
  }
  return str.repeat(num);
}


// Using num with a counter
function repeatStringNumTimes(str, num) {
  "use strict";
  var storageArray = [];
  var count = 0;

  // if num is not a positive number
  if (num < 0) {
    console.log("This test value is < 0...so returned an empty string");
    return "";
  }

  while (count < num) {
      storageArray.push(str);
      count += 1;
  }
  var repeatedStr = storageArray.join('');
  console.log(repeatedStr);
}


repeatStringNumTimes("*", 3); // "***"
repeatStringNumTimes("abc", 3); // "abcabcabc"
repeatStringNumTimes("abc", 4); // "abcabcabcabc".
repeatStringNumTimes("abc", 1); // "abc".
repeatStringNumTimes("*", 8); // "********".
repeatStringNumTimes("abc", -2); // ""



// JavaScript 6

// Using .repeat method but with ternary operator
const repeatStringNumTimesES6 = (str, num) => num < 0 ? "" : str.repeat(num)


repeatStringNumTimesES6("*", 3); // "***"
repeatStringNumTimesES6("abc", 3); // "abcabcabc"
repeatStringNumTimesES6("abc", 4); // "abcabcabcabc".
repeatStringNumTimesES6("abc", 1); // "abc".
repeatStringNumTimesES6("*", 8); // "********".
repeatStringNumTimesES6("abc", -2); // ""

File: repeat_a_string/repeat_a_string.py
"""
Repeat a given string (first argument) num times (second argument). Return an empty string if num is not a positive number.


>>> repeat_string_num_times("*", 3)
'***'
>>> repeat_string_num_times("abc", 3)
'abcabcabc'
>>> repeat_string_num_times("hey", 4)
'heyheyheyhey'
>>> repeat_string_num_times("abc", 1)
'abc'
>>> repeat_string_num_times("*", 8)
'********'
>>> repeat_string_num_times("abc", -2)
' '

"""

# example 1 - Using 2nd parameter as multiplier
def repeat_string_num_times(string: str, num: int) -> str:
    if num < 0:
        return " "
    else:
        return string * num

# example 2 - Using 2nd parameter with a counter
def repeat_string_num_times(string: str, num: int) -> str:
    count = 0
    new_str = ""

    if num < 0:
        return " "

    while count < num:
        new_str += string
        count += 1
    return new_str



